### DISCUSSION TITLE
[Feature Request] CC:Tweaked integration

### DISCUSSION THREAD

**Discussion by vico93** [ACGaming/UniversalTweaks#340] [Tweak Suggestions]
Again, don't know if such suggestions are within the scope of UniversalTweaks, but CC need some facelift to allow fully turtle automation and more efficient use of turtles.

The points:
- Allow turtles to equip any tier by default (i know its possible to define custom tier other than diamond using Craftweaker, but it annoying and doesn't carry the damage of each tier - next item);
- Enable durability and breakable blocks: by default, the equipped diamond turtle/sword/shovel/hoe doesn't break after the X uses of durability the same item norrmally do when equipped by a player. So add this feature and in case of implementing the first point, add the durability mechanic of each tier;
- (OPTIONAL) Separate the actions equivalent to left click (generally attack/break block) and right click (for multi-use tool can be either till/make grass path/strip log depending of the tool) in different functions instead of both under `turtle.dig()`. It's a remmant of the older versions when CC started being developed and can cause headaches when handling axes/hoes/shovels on Lua. Maybe make `dig()` exclusively do the block break, while add an `use()` for the right-click actions;
- (OPTIONAL) Allow equipping of bows and fishing rods, and in those tools the `use()` functions (if the above topic gets worked on as well) should get angle and strenght parameters to shoot the arrow/bob forward, and in the rod case `use()` again would pull the bob.
- Allow turtles to detect entities using `detect()` method; this is essential to make automatic guards to not accidentally attack players or tamed mobs like wolves;
- Add a toggleable option to allow turtles (at least some of them, configurable via script, or just advanced ones) to keep the chunks they are in loaded. Needing to go after several turtles to place chunk loaders is annoying and also slows considerably any work that could be carried automatically by the turtles if this feature would already exist.

**Comment by Krutoy242** [comment]
You just described default features of OpenComputers 😅
